packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4369,67 +4369,343 @@ class PositionedDirectional extends StatelessWidget {
+/// min and max, for both width and height.
+///
+/// There is another case in which there are no constraints given,
+/// and this can be the case with the children of a [Flex].
+/// This is also the source of many errors seen while using [Row]s
+/// and [Column]s. Error messages referring to this will often use
+/// the term _unbounded_ to describe the situation in which no
+/// constraints (min or max size limits) have been passed down from
+/// the parent.
+///
+/// Now that all of those terms have been explained, the multi-child
+/// layout algorithm for a [Flex] proceeds in six steps:
+///
+/// 1. Layout each child that is not a [Flexible] or an [Expanded],
+///    but do so without setting any minimum or maximum size in the
+///    main axis. This is called laying the child out "without applying
+///    any constraints in the main axis".
+///
+///    Apply any constraints in the _cross axis_ direction that may have
+///    been passed down by the parent of the [Flex], without forcing the
+///    child to be any larger than it's stated size in that direction. This is
+///    called using _"loose constraints"_ (in this case, in the cross axis).
+///
+///    However, if the [CrossAxisAlignment] of the [Flex] has been
+///    set to [CrossAxisAlignment.stretch] then force all the children of the
+///    [Flex] to be as large as possible in the cross axis direction. This is
+///    called using _"tight constraints"_.
+///
+/// 2. In the main axis, take whatever space is left over after
+///   laying out everything that is not a [Flexible] and divide that
+///   space among the [Flexible]s. (Remember, [Expanded] is just
+///   a kind of [Flexible]).
+///
+///   The [Flexible.flex] parameter determines how much of the
+///   leftover space each [Flexible] will end up having reserved
+///   for it. Notice that _"reserved for it"_ does not mean how big
+///   it's going to be. It means the space (or size) reserved for it
+///   is the maximum size the [Flexible] _could be,_ in the main axis.
+///
+///   Add up all the [flex] values of each [Flexible]'s fit parameter,
+///   and each [Flexible] will have a percentage of the leftover space
+///   reserved for it. This percentage will be equal to the percentage
+///   of total [flex] each [Flexible]'s [flex] value represents. IE:
+///
+///   [Flexible] 1 has [flex] of 50.
+///   [Expanded] 1 has [flex] of 30.
+///   [Expanded] 2 has [flex] of 20.
+///
+///   In the above case, the total of all flex parameters is 100. The
+///   [Flexible] will get 50/100 (50%) of the remaining space reserved
+///   for it. The first [Expanded] will get 30/100 (30%) reserved for it,
+///   and the second one will get 20% of the remaining space
+///   reserved for it.
+///
+///   The numbers you use for your [Flexible.fit] parameters
+///   don't have to add to 100. This example does so only because
+///   it makes understanding the process easier.
```

Comment:
```suggestion
///   don't have to add to 100, they will be proportional to their sum.
```
There's no need to have commentary about why you're talking about things, it just distracts from the message.